### PR TITLE
Refactor LXD API operations

### PR DESCRIPTION
**Pull request by zulcss**
* Consolidate container_utils and container_client into one class. 
* Split out snapshot, container, image, migrate, event, and host operations into their own  mixin
  classes.
* This makes the code easier to follow and to spot bugs easier and allow for optimizations.
* This is the first step to convert using taskflow.

### CODE REVIEW

File: nova_lxd/nova/virt/lxd/client/client.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#    under the License.
+
+from nova_lxd.nova.virt.lxd.client import container
+from nova_lxd.nova.virt.lxd.client import event
+from nova_lxd.nova.virt.lxd.client import image
+from nova_lxd.nova.virt.lxd.client import migrate
+from nova_lxd.nova.virt.lxd.client import snapshot
+
+
+class Client(container.MixinContainer,
```
Comment (rockstar) [inline nova_lxd/nova/virt/lxd/client/client.py]: This class needs a docstring. Why is it here? What does it provide? How is it used? Once you have the docstring, you also won't need the empty `__init__` method.

File: nova_lxd/nova/virt/lxd/client/container.py [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+#    under the License.
+
+from nova.compute import power_state
+from nova import exception
+from nova import i18n
+from pylxd import exceptions as lxd_exceptions
+
+from oslo_config import cfg
+from oslo_log import log as logging
+from oslo_utils import excutils
```
Comment (rockstar) [inline nova_lxd/nova/virt/lxd/client/container.py]: These are all third party imports, correct? They should be alphabetized, so these `oslo_*` imports should go before `pylxd` imports.

File: nova_lxd/nova/virt/lxd/client/container.py [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+
+_ = i18n._
+_LI = i18n._
+_LE = i18n._
+
+CONF = cfg.CONF
+LOG = logging.getLogger(__name__)
+
+
+class MixinContainer(object):
```
Comment (rockstar) [inline nova_lxd/nova/virt/lxd/client/container.py]: A docstring here is also necessary, methinks.

File: nova_lxd/nova/virt/lxd/client/event.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+from nova_lxd.nova.virt.lxd.client import host
+
+_ = i18n._
+_LE = i18n._LE
+
+CONF = cfg.CONF
+LOG = logging.getLogger(__name__)
+
+
+class MixinEvent(object):
```
Comment (rockstar) [inline nova_lxd/nova/virt/lxd/client/event.py]: Same here for a docstring.

File: nova_lxd/nova/virt/lxd/client/host.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+from oslo_log import log as logging
+
+_ = i18n._
+
+CONF = cfg.CONF
+CONF.import_opt('host', 'nova.netconf')
+LOG = logging.getLogger(__name__)
+
+
+class Host(object):
```
Comment (rockstar) [inline nova_lxd/nova/virt/lxd/client/host.py]: Another decstring.

File: nova_lxd/nova/virt/lxd/client/image.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+from nova_lxd.nova.virt.lxd.client import host
+
+_ = i18n._
+_LE = i18n._
+
+CONF = cfg.CONF
+LOG = logging.getLogger(__name__)
+
+
+class MixinImage(object):
```
Comment (rockstar) [inline nova_lxd/nova/virt/lxd/client/image.py]: Another docstring.

File: nova_lxd/nova/virt/lxd/client/migrate.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+from nova_lxd.nova.virt.lxd.client import host
+
+_ = i18n._
+_LE = i18n._LE
+
+CONF = cfg.CONF
+LOG = logging.getLogger(__name__)
+
+
+class MixinMigrate(object):
```
Comment (rockstar) [inline nova_lxd/nova/virt/lxd/client/migrate.py]: Docstring... :smile:

File: nova_lxd/nova/virt/lxd/client/snapshot.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+_ = i18n._
+_LE = i18n._LE
+
+CONF = cfg.CONF
+CONF.import_opt('host', 'nova.netconf')
+LOG = logging.getLogger(__name__)
+
+
+class MixinSnapshot(object):
```
Comment (rockstar) [inline nova_lxd/nova/virt/lxd/client/snapshot.py]: Another need for a docstring.

**Comment by rockstar** [comment]
This is a large amount of code that is now being tested but isn't being used anywhere, and that would indicate that I'm missing some context. Could you expand a bit on what that means?

When writing unit tests (or tests in general), docstrings are an absolutely crucial part of the test. The prose describing the test is just as important as the test itself. Jono Lange wrote a very good post on this that I refer to often: https://jml.io/pages/test-docstrings.html

**Comment by zulcss** [comment]
Better version is coming..

**Pull request closed without merging**